core/src/main/java/org/apache/calcite/rel/metadata/RelMdRowCount.java

```diff
@@ -108,7 +108,11 @@ public Double getRowCount(Intersect rel, RelMetadataQuery mq) {
         rowCount = partialRowCount;
       }
     }
-    return rowCount;
+    if (!rel.all) {
+      return rowCount;
+    } else {
+      return rowCount * 2;
+    }
```

Comment:
If rowCount is null, it would throw NPE.

Reply:
Thanks, update it.